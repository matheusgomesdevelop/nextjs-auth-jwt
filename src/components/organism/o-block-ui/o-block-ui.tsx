import { BlockUIProps, BlockUI } from "primereact/blockui";

export interface I_BlockUI extends BlockUIProps {}

export const O_BlockUI = ({ ...props }: I_BlockUI) => {
  return <BlockUI {...props} />;
};
