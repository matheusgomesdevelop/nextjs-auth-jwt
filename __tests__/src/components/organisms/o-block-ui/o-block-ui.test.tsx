import renderer from "react-test-renderer";

// Organism
import { O_BlockUI } from "@/components/organism/o-block-ui/o-block-ui";

describe("Deve renderizar o block ui corretamente", () =>
  it("Deve preservar a estrutura visual do componente", () => {
    const three = renderer.create(<O_BlockUI icon="pi-spin pi-cog" />).toJSON();

    expect(three).toMatchSnapshot();
  }));
