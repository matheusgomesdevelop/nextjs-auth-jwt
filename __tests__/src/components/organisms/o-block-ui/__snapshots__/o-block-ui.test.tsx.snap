// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o block ui corretamente Deve preservar a estrutura visual do componente 1`] = `
<div
  className="p-blockui-container"
  data-pc-name="blockui"
  data-pc-section="root"
  icon="pi-spin pi-cog"
  id={null}
  style={{}}
/>
`;
